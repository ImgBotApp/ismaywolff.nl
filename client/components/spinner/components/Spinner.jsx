import { css } from 'glamor'
import glamorous from 'glamorous'
import { colors } from '../../../styles'

const scale = css.keyframes({
  '0%': { transform: 'scale(0)' },
  '100%': { transform: 'scale(1)', opacity: 0 }
})

const Spinner = glamorous.div({
  width: '2rem',
  height: '2rem',
  margin: '2rem auto',
  backgroundColor: colors.black,
  borderRadius: '100%',
  animation: `${scale} 1s infinite ease-in-out`
})

export default Spinner
