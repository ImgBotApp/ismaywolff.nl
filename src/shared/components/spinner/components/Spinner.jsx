import styled, { keyframes } from 'styled-components'
import { colors } from '../../../styles'

const scale = keyframes`
  from {
    transform: scale(0);
  }

  to {
    opacity: 0;
    transform: scale(1);
  }
`

const Spinner = styled.div`
  animation: ${scale} 1s infinite ease-in-out;
  background-color: ${colors.black};
  border-radius: 100%;
  height: 2rem;
  margin: 2rem auto;
  width: 2rem;
`

export default Spinner
