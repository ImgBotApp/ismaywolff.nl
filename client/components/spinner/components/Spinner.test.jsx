import React from 'react'
import { shallow } from 'enzyme'
import { matcher, serializer } from 'jest-glamor-react'
import Spinner from './Spinner'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('<Spinner />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Spinner />)
    expect(wrapper).toMatchSnapshotWithGlamor()
  })
})
