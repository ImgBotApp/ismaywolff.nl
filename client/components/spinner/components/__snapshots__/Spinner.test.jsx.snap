// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders correctly 1`] = `
<div
  className="css-ae4b0b"
/>
`;
